src/simulator/views/gui/view_editor.py

```diff
@@ -620,153 +712,66 @@ def __init__(self, services: Services, voxel_map: VoxelMap):
             pressEffect=1,
-            command=lambda: self.load_state(self.state_num),
+            command=self.__reset,
             pos=(0.50, 0, -5.5),
             parent=self.__window.frame,
             scale=(0.20, 2.1, 0.15),
             frameColor=TRANSPARENT)
 
-        # view elements
-        self.__elements = []
-        for name, dc in self.__voxel_map.colours.items():
-            self.__elements.append(ViewElement(name, self.__voxel_map.set_visibility, self.__voxel_map.set_colour, self.__window.frame, colour=dc()))
-
-        for i in range(len(self.__elements)):
-            self.__elements[i].frame.set_pos((0.0, 0.0, -1.9 - 0.2 * i))
-
-        self.__cv_transparent_overlays = []
-        for i in range(len(self.__elements)):
-            self.__cv_transparent_overlays.append(DirectButton(parent=self.__window.frame,
-                                                               relief=DGG.SUNKEN,
-                                                               frameColor=TRANSPARENT,
-                                                               borderWidth=(0, 0),
-                                                               frameSize=(-0.52, 0.52, -0.44, 0.44),
-                                                               scale=(0.18, 1.0, 0.18),
-                                                               pos=(-0.65, 1.0, -1.897 - 0.2 * i),
-                                                               command=self.__select_cv,
-                                                               extraArgs=[i]))
-
-        # Creating state buttons
-        self.__state_btns = []
+        # Creating view selectors
+        self.__view_selectors = []
         for i in range(0, 3):
             num = os.path.join(DATA_PATH, str(i + 1) + ".png")
 
-            self.__state_btns.append(DirectButton(image=num,
-                                                  pos=(-0.7 + i * 0.7, 0.4, -4.4),
-                                                  parent=self.__window.frame,
-                                                  scale=0.16,
-                                                  frameColor=TRANSPARENT,
-                                                  command=self.__select_state_num_one,
-                                                  extraArgs=[i]))
+            self.__view_selectors.append(DirectButton(image=num,
+                                                      pos=(-0.7 + i * 0.7, 0.4, -4.4),
+                                                      parent=self.__window.frame,
+                                                      scale=0.16,
+                                                      frameColor=TRANSPARENT,
+                                                      command=lambda v: setattr(self, "view_idx", v),
+                                                      extraArgs=[i]))
         for i in range(0, 3):
             num = os.path.join(DATA_PATH, str(i + 4) + ".png")
```

Comment:
These two `for` loops are a bit hacky - if it's for the coordinates, you can use `-0.7 + (i%3) * 0.7` for the x and `-4.4 - 0.5 * (i // 3)` for the y, and just use `range(0, 6)`

Reply:
Done with some further generalisation